internal/resources/connections/resource_metrics_endpoint_scrape_job.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common"
+	"github.com/grafana/terraform-provider-grafana/v3/internal/common/connectionsapi"
+)
+
+var (
+	resourceMetricsEndpointScrapeJobTerraformName = "grafana_connections_metrics_endpoint_scrape_job"
+	resourceMetricsEndpointScrapeJobTerraformID   = common.NewResourceID(common.StringIDField("stack_id"), common.StringIDField("name"))
+)
+
+type resourceMetricsEndpointScrapeJob struct {
+	client *connectionsapi.Client
+}
+
+func makeResourceMetricsEndpointScrapeJob() *common.Resource {
+	return common.NewResource(
+		common.CategoryConnections,
+		resourceMetricsEndpointScrapeJobTerraformName,
+		resourceMetricsEndpointScrapeJobTerraformID,
+		&resourceMetricsEndpointScrapeJob{},
+	)
+}
+
+func (r *resourceMetricsEndpointScrapeJob) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Configure is called multiple times (sometimes when ProviderData is not yet available), we only want to configure once
+	if req.ProviderData == nil || r.client != nil {
+		return
+	}
+
+	client, err := withClientForResource(req, resp)
+	if err != nil {
+		return
+	}
+
+	r.client = client
+}
+
+func (r *resourceMetricsEndpointScrapeJob) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = resourceMetricsEndpointScrapeJobTerraformName
+}
+
+func (r *resourceMetricsEndpointScrapeJob) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The Terraform Resource ID. This has the format \"{{ stack_id }}:{{ name }}\".",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					// See https://developer.hashicorp.com/terraform/plugin/framework/resources/plan-modification#usestateforunknown
+					// for details on how this works.
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"stack_id": schema.StringAttribute{
+				Description: "The Stack ID of the Grafana Cloud instance. Part of the Terraform Resource ID.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the metrics endpoint scrape job. Part of the Terraform Resource ID.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"enabled": schema.BoolAttribute{
+				Description: "Whether the metrics endpoint scrape job is enabled or not.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(true),
+			},
+			"authentication_method": schema.StringAttribute{
```

Comment:
is this field necessary? you could do a resource-level validation to ensure that either both `authentication_bearer_token` and `authentication_basic_username` are set, or `authentication_basic_password` is set, but not both kinds. See https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators@v0.14.0/resourcevalidator

Reply:
Thanks for the suggestion, I've added in resource validation here aa0e52d12e5265f74c7d8e1c9a5d30afc3b47e86 so that we can catch these misconfigurations at plan time.

Re is the field necessary - the API does in deed expect it. If we remove it from the schema, we'd have to add some more logic in the client to re-populate it. Ultimately this is moving us away from our goal of having the terraform implementation a thin layer on top of our API spec without any bespoke logic.
I think in the future, we can look at removing this field from the job API entirely, and then subsequently from here - but think the validation resolves many of the possible paper cuts here for now. 
Let me know if it makes sense to you.

Reply:
I'll ultimately leave the decision up to you in terms of your goals etc. My suggestion came from perceiving some overlap in the fields that a user must configure to use the provider. Removing overlap on the Terraform side would make user configuration simpler at the cost of, as you say, complicating the provider implementation vs your goal of implementing a thin layer over the API. It sounds like the validation suggestion still benefits, which is great, and in the future, as you say, you may just remove the field from the API altogether, which sounds like a good plan.

Reply:
Will resolve this and leave for a followup action as we can take this out without it being a breaking change.